app/services/line_item_syncer.rb

```diff
@@ -23,19 +23,29 @@ def update_item_quantities(order)
     changed_subscription_line_items.each do |sli|
       line_item = order.line_items.find_by_variant_id(sli.variant_id)
       next if update_quantity(line_item, sli)
-      product_name = "#{line_item.product.name} - #{line_item.full_name}"
-      order_update_issues.add(order, product_name)
+
+      add_order_update_issue(order, line_item)
     end
   end
 
   def create_new_items(order)
     new_subscription_line_items.each do |sli|
-      order.line_items.create(variant_id: sli.variant_id, quantity: sli.quantity, skip_stock_check: true)
+      new_line_item = order.line_items.create(variant_id: sli.variant_id,
+                                              quantity: sli.quantity,
+                                              skip_stock_check: skip_stock_check?(order))
+      next if skip_stock_check?(order) || new_line_item.sufficient_stock?
+
+      add_order_update_issue(order, new_line_item)
+      new_line_item.update_attributes(quantity: 0)
```

Comment:
because this syncing business is tricky... after adding the item to the subs, the user will want to adapt the quantity right? if that happens, the item will not be in the order and the code would blow up.
but I went for it now and fixed this case.
I switched back to deleting the line item as you expected and then I had to adapt the update quantity logic for the case where the item is added and then quantity updated. I think it's good now, the user is notified the item is not added in the first place and then when quantity is updated the user is again notified that the order was not updated.